roles/dns_server/tasks/main.yml

```diff
@@ -0,0 +1,40 @@
+---
+- name: Get OpenShift facts
+  openshift_facts:
+    local_facts:
+      hostname: "{{ openshift_hostname | default(None) }}"
+      ip: "{{ openshift_ip | default(None) }}"
+      public_hostname: "{{ openshift_public_hostname | default(None) }}"
+      public_ip: "{{ openshift_public_ip | default(None) }}"
+
+- name: Calculate subnet
+  set_fact: subnet={{ openshift.common.ip|subnet }}
+  run_once: true
+
+- name: Set subnet name
+  set_fact: subnet_name={{ subnet|subnet_name }}
+  run_once: true
+
+- name: Set hostname
+  hostname: name={{ openshift.common.hostname }}
+
+- name: Install bind DNS server
+  yum: name=bind state=latest
+
+  # FIXME: this should be generated, not copied
+- name: Copy rndc key
+  copy: src=rndc.key dest=/etc/rndc.key mode=0640 owner=root group=named
```

Comment:
See https://github.com/openshift/openshift-extras/blob/enterprise-2.2/enterprise/install-scripts/generic/openshift.sh#L2895 for how we did this in OSE 2.2

The ansible command module has a creates parameter that will only run the file does not already exist: http://docs.ansible.com/command_module.html